core/src/main/java/com/graphhopper/routing/Router.java

```diff
@@ -303,7 +305,7 @@ private ResponsePath concatenatePaths(GHRequest request, Weighting weighting, Qu
     }
 
     private PointList getWaypoints(List<Snap> snaps) {
-        PointList pointList = new PointList(snaps.size(), true);
+        PointList pointList = new PointList(snaps.size(), graph.getNodeAccess().is3D());
```

Comment:
This was a bit of an inconsistency, where we used a 3D pointlist even without elevation, but it does not change the server-response, as the ResponsePathSerializer already took care of this.